components/microsoft_onedrive/sources/new-file-in-folder/new-file-in-folder.js

```diff
@@ -0,0 +1,46 @@
+const base = require("../new-file/new-file");
```

Comment:
Can you walk me through the logic of requiring the `new_file` component instead of keeping the common logic in `base` and requiring that?

Reply:
The `new-file-in-folder` component is a refinement over `new-file`:
- It also scans the drive for new files. This logic is implemented entirely in the `new-file` component and it can be completely reused by the `new-file-in-folder` component
- It adds some logic on top of the "new file" event source logic:
  - Accept an additional user prop (i.e. the "Folder" prop)
  - When a change happens in the drive, only scan for files under the specified folder

So the hierarchy would be something like this:
![image](https://user-images.githubusercontent.com/5479513/115972879-f0e6a900-a505-11eb-9f3d-0274fe89ab2c.png)